### PR TITLE
fix(client) fix self signed certs on Chrome

### PR DESCRIPTION
**Pull request by cintolas**
- [x] commit message named according to [Contributing Guide](https://github.com/coderaiser/cloudcmd/blob/master/CONTRIBUTING.md "Contributting Guide")
- [ ] `npm run fix:lint` is OK
- [x] `npm test` is OK

### CODE REVIEW

**Comment by coderaiser** [comment]
Could you please add test for this.

**Comment by cintolas** [comment]
I tried to add a test.  However there is an issueThe test itself “fails” or passes depending on how you see interpret this.The issue is that this is a chrome specific security issue. All other browsers including the test script browser in the repo  that I have tested allow self signed certs. If you have guidance on how you want to proceed I will add what you need. Sent from my iPhoneOn May 12, 2023, at 7:00 AM, coderaiser ***@***.***> wrote:﻿
Could you please add test for this.

**Comment by coderaiser** [comment]
Sure, here is test file https://github.com/coderaiser/cloudcmd/blob/master/client/sw/register.spec.js#L9

[mock-require](https://www.npmjs.com/package/mock-require) used for mocking.

**Comment by cintolas** [comment]
updated test to test self-signed

File: client/sw/register.js [hunk 1]
```diff
@@ -17,7 +19,12 @@ async function registerSW(prefix) {
     if (!isHTTPS && !isLocalhost)
         return;
     
-    return await navigator.serviceWorker.register(`${prefix}/sw.js`);
+    const [e, sw] =  await tryToCatch(navigator.serviceWorker.register,`${prefix}/sw.js`);
+    
+    if(e)
+        return;
```
Comment (coderaiser) [inline client/sw/register.js:25]:
```suggestion
        return null;
```

File: client/sw/register.js [hunk 2]
```diff
@@ -17,7 +19,12 @@ async function registerSW(prefix) {
     if (!isHTTPS && !isLocalhost)
         return;
     
-    return await navigator.serviceWorker.register(`${prefix}/sw.js`);
+    const [e, sw] =  await tryToCatch(navigator.serviceWorker.register,`${prefix}/sw.js`);
+    
+    if(e)
```
Comment (coderaiser) [inline client/sw/register.js:24]:
```suggestion
    if (e)
```

File: client/sw/register.spec.js [hunk 3]
```diff
@@ -70,6 +70,23 @@ test('sw: register: registerSW: http', async (t, {location, navigator}) => {
...
+        hostname: 'self-signed.badssl.com',
+    });
+    
+    const {register} = navigator.serviceWorker;
+    
+    const {registerSW} = reRequire('./register');
+    
+    await registerSW();
+    
+    t.pass('should not call register');
```
Comment (coderaiser) [inline client/sw/register.spec.js:85]: t.notOk(result, 'should not throw');

Comment (coderaiser) [inline client/sw/register.spec.js:85]:
```suggestion
    t.notOk(result, 'should not throw');
```

File: client/sw/register.spec.js [hunk 4]
```diff
@@ -70,6 +70,23 @@ test('sw: register: registerSW: http', async (t, {location, navigator}) => {
     t.end();
 });
 
+test('sw: register: registerSW: https self-signed', async (t, {location, navigator}) => {
+    Object.assign(location, {
+        protocol: 'https',
+        hostname: 'self-signed.badssl.com',
+    });
+    
+    const {register} = navigator.serviceWorker;
```
Comment (coderaiser) [inline client/sw/register.spec.js:79]:
```suggestion
    const {register} = navigator.serviceWorker;
    register.throws(Error('Cannot register service worker!'));
```

https://github.com/cloudcmd/stub#stubthrowserror

**Comment by coderaiser** [comment]
@cintolas nice! Add a couple updates to tests, so if register of service worker throws, we should receive some kind of result anyways.

Have you tried to run this code in your browser?

**Pull request merged**